Service/SalesReportService.php

```diff
@@ -134,146 +217,297 @@ private function convert($data)
         }
 
-        return array(
-            'raw' => $raw,
-            'graph' => array(
-                'labels' => array_keys($price),
-                'datasets' => array(
-                    array(
-                        'label' => "購入金額",
-                        'fillColor' => 'rgba(255,255,255,0.0)',
-                        'strokeColor' => 'rgba(151,187,205,1)',
-                        'pointColor' => 'rgba(151,187,205,1)',
-                        'pointStrokeColor' => '#fff',
-                        'pointHighlightFill' => '#fff',
-                        'pointHighlightStroke' => 'rgba(151,187,205,1)',
-                        'data' => array_values($price),
-                    ),
-                ),
-            )
+        log_info('SalesReport Plugin : term report ', array('result count' => $orderNumber));
+        // Return null and not display in screen
+        if ($orderNumber == 0) {
+            return array(
+                'raw' => null,
+                'graph' => null,
+            );
+        }
+
+        $graph = array(
+            'labels' => array_keys($price),
+            'datasets' => array(
+                'label' => '購入合計',
+                'data' => array_values($price),
+                'lineTension' => 0.1,
+                'backgroundColor' => 'rgba(75,192,192,0.4)',
+                'borderColor' => 'rgba(75,192,192,1)',
+                'borderCapStyle' => 'butt',
+                'borderDash' => array(),
+                'borderDashOffset' => 0.0,
+                'borderJoinStyle' => 'miter',
+                'pointBorderColor' => 'rgba(75,192,192,1)',
+                'pointBackgroundColor' => '#fff',
+                'pointBorderWidth' => 1,
+                'pointHoverRadius' => 5,
+                'pointHoverBackgroundColor' => 'rgba(75,192,192,1)',
+                'pointHoverBorderColor' => 'rgba(220,220,220,1)',
+                'pointHoverBorderWidth' => 2,
+                'pointRadius' => 1,
+                'pointHitRadius' => 10,
+                'spanGaps' => false,
+                'borderWidth' => 1,
+            ),
         );
-    }
 
-    private function formatUnit()
-    {
-        $unit = array(
-            'byDay' => 'm/d',
-            'byMonth' => 'm',
-            'byWeekDay' => 'D',
-            'byHour' => 'H',
+        return array(
+            'raw' => $raw,
+            'graph' => $graph,
         );
-        return $unit[$this->unit];
     }
 
+    /**
+     * product sale report.
+     *
+     * @param array $data
+     *
+     * @return array
+     */
     private function convertByProduct($data)
     {
+        $i = 0;
+        $label = array();
+        $graphData = array();
+        $backgroundColor = array();
         $products = array();
+        /* @var $entityManager EntityManager */
+        $entityManager = $this->app['orm.em'];
+        $sql = 'Select od.product_class_id From dtb_order_detail od Where od.order_detail_id = :order_detail_id';
```

Comment:
With soft delete filter, I can not access customer/product class to retrieve the data was deleted.
I just get data via raw query.
Do you have any idea for this?

[Source](https://github.com/EC-CUBE/ec-cube/blob/master/src/Eccube/Resource/doctrine/Eccube.Entity.Order.dcm.yml#L178)

I had questions in [here](https://github.com/EC-CUBE/lockonvn-team/issues/16)